### PR TITLE
Try (once) to reconnect after an IMAP failure

### LINKED ISSUE #346: IMAP: mailbox closed
I am enthusiastic to see an actively maintained fork of mutt.
Unfortunately the ages old issue with random closing of IMAP mailboxes is still present here.

http://mutt-users.mutt.narkive.com/gy14Cvl9/mailbox-closed-mutt-behaviour

Can neomutt be configured to connect every n minutes and during mailbox syncing (`$`) rather than maintaining active connection at all times that gets randomly killed?

### PR DESCRIPTION
**Pull request by gahr**
Closes #346

### CODE REVIEW

**Comment by gahr** [comment]
@neomutt/reviewers Please review

File: imap/command.c [hunk 1]
```diff
@@ -440,6 +440,7 @@ static int cmd_status (const char *s)
 /* cmd_handle_fatal: when IMAP_DATA is in fatal state, do what we can */
 static void cmd_handle_fatal (IMAP_DATA* idata)
 {
+  static int try_reconnect = 1;
```
Comment (d-k-c) [inline imap/command.c]: Could we keep the function reentrant by moving this to IMAP_DATA? Otherwise 👍

**Comment by gahr** [comment]
> Kill it again and the mailbox is left closed.

@flatcap that is surprising... the "once" in the comment was meant to be "once per failure", that is, if mutt can't re-establish a connection on failure, it doesn't try again. That's the reason for the `try_connect` static var.

If it doesn't reconnect upon a second (unrelated, long after reconnection) failure, then there's still something I have to look at.

File: imap/command.c (hunk 1 above)
Comment (gahr) [inline imap/command.c]: The function *is* reentrant, just it doesn't behave the same once reentred ;) I don't see a benefit to moving this local piece of data to IMAP_DATA. What am I missing?

**Pull request merged**